drivers/shared/executor/executor_linux.go

```diff
@@ -195,9 +197,30 @@ func (l *LibcontainerExecutor) Launch(command *ExecCommand) (*ProcessState, erro
 	l.systemCpuStats = stats.NewCpuStats()
 
 	// Starts the task
-	if err := container.Run(process); err != nil {
-		container.Destroy()
-		return nil, err
+	if command.NetworkIsolation != nil && command.NetworkIsolation.Path != "" {
+		// Lock to the thread we're changing the network namespace of
+		runtime.LockOSThread()
+		netns, err := ns.GetNS(command.NetworkIsolation.Path)
+		if err != nil {
+			return nil, err
+		}
+
+		// Start the container in the network namespace
+		err = netns.Do(func(ns.NetNS) error {
+			if err := container.Run(process); err != nil {
+				container.Destroy()
```

Comment:
Q: Does the destroy also need to happen in the netns.Do context?  If so, we need to wrap other `container.Destroy` calls?  If not, we probably can reduce if statements for error handling like above.